sdk/core/azure-core/src/credentials/credentials.cpp

```diff
@@ -49,8 +49,9 @@ AccessToken Azure::Core::Credentials::ClientSecretCredential::GetToken(
   static std::string const errorMsgPrefix("ClientSecretCredential::GetToken: ");
   try
   {
-    Http::Url url;
-    url.SetPath(m_authority + UrlEncode(m_tenantId) + "/oauth2/v2.0/token");
+    Http::Url url(m_authority);
+    url.AppendPath(UrlEncode(m_tenantId));
```

Comment:
maybe don't need to encode

Reply:
.NET MSAL does it, but differently. And in most cases, encoded and non-encoded tenant ID look the same, if we are talking about the ones you can realistically have. I'd say, at this point whether you do `UrlEncode(m_tenantId)` or not, it shouldn't matter much. Previously, it was working with us doing the encoding. But on the other hand, no code was using it at a time. Plus, it is always easier to not encode, and tell the client code to encode.